kuma/wiki/jobs.py

```diff
@@ -8,8 +9,15 @@
 
 
 class DocumentNearestZoneJob(KumaJob):
-    lifetime = 60 * 60 * 29
-    refresh_timeout = 60
+    # Allow up to three minutes to refresh the cache before assuming
+    # the task has failed and allowing another to be enqueued.
+    refresh_timeout = 180
+
+    @property
+    def lifetime(self):
+        # Spread the cache expiration times across a random
+        # number of days from 1 to 10 (in units of seconds).
+        return random.randint(1, 10) * 24 * 60 * 60
```

Comment:
I had originally intended on overriding ``expiry``, but decided on this approach thinking that it only changes what we intend. I had originally went through the same logic you outline in deciding that, but now on second thought, if ever ``fetch_on_stale_threshold`` changes to something other than ``None``, the random value of ``lifetime`` becomes problematic. So I think it's safer to just override ``expiry`` so that if ``fetch_on_stale_threshold`` ever changes to something other than ``None``, a constant value for ``lifetime`` allows more control over [should_stale_item_be_fetched_synchronously](https://github.com/codeinthehole/django-cacheback/blob/master/cacheback/base.py#L384-L391) without overriding it as well.